### PR TITLE
API-1090: Add an UnknownAttributeException

### PR DESCRIPTION
**Pull request by pchasle**
Add an `UnknownAttributeException` (in place of `UnknownPropertyException`) to identify more precisely this kind of error.

```
Akeneo\Pim\Enrichment\Component\Product\Exception\UnknownAttributeException
```

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php [hunk 1]
```diff
@@ -526,15 +526,6 @@ protected function filterEmptyValues(ProductInterface $product, array $data): ar
...
             }
-        } catch (UnknownPropertyException $exception) {
-            throw new DocumentedHttpException(
-                Documentation::URL . 'patch_products__code_',
-                sprintf(
-                    'Property "%s" does not exist. Check the expected format on the API documentation.',
-                    $exception->getPropertyName()
-                ),
-                $exception
-            );
```
Comment (pchasle) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php:537]: Redundant, the `UnknownPropertyException` already display this message and is handled by the `catch PropertyException` below to create the `DocumentedHttpException`.

File: tests/back/Pim/Enrichment/Specification/Component/Product/Comparator/Filter/EntityWithValuesFilterSpec.php [hunk 2]
```diff
@@ -227,6 +228,36 @@ function it_throws_an_exception_if_code_is_not_found(
             );
     }
 
+    function it_throws_an_unknown_attribute_exception_if_an_attribute_code_is_not_found(
+        $normalizer,
+        $attributeRepository,
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Comparator/Filter/EntityWithValuesFilterSpec.php:233]:
```suggestion
        NormalizerInterface $normalizer,
        AttributeRepositoryInterface $attributeRepository,
```

**Comment by mmetayer** [comment]
Looks good, you probably forgot to update error messages in API e2e test though

**Comment by mmetayer** [comment]
Another question: don't you want to apply this change to the ProductModelUpdater as well?

**Comment by pchasle** [comment]
> Another question: don't you want to apply this change to the ProductModelUpdater as well?

We might do this in another PR (with the same logic). Unless you think we should ship the two together?

**Pull request merged**